sdk/typescript/test/e2e/dynamic-fields.test.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect, beforeAll } from 'vitest';
+import {
+  LocalTxnDataSerializer,
+  RawSigner,
+} from '../../src';
+import {
+  publishPackage,
+  setup,
+  TestToolbox,
+} from './utils/setup';
+
+describe.each([{ useLocalTxnBuilder: true }])(
+  'Dynamic Fields Reading API',
+  ({ useLocalTxnBuilder }) => {
+    let toolbox: TestToolbox;
+    let signer: RawSigner;
+    let packageId: string;
+    let shouldSkip: boolean;
+
+    beforeAll(async () => {
+      toolbox = await setup();
+      const version = await toolbox.provider.getRpcApiVersion();
+      shouldSkip = version?.major == 0 && version?.minor < 20;
+      signer = new RawSigner(
+        toolbox.keypair,
+        toolbox.provider,
+        useLocalTxnBuilder
+          ? new LocalTxnDataSerializer(toolbox.provider)
+          : undefined
+      );
+      const packagePath = __dirname + '/./data/dynamic_fields';
+      packageId = await publishPackage(signer, useLocalTxnBuilder, packagePath);
+    });
+
+    it('get all dynamic fields', async () => {
+
+      let obj_owned_by_address = await toolbox.provider.getObjectsOwnedByAddress(toolbox.address());
```

Comment:
```suggestion
      const objs_owned_by_address = await toolbox.provider.getObjectsOwnedByAddress(toolbox.address());
```
nit: plural + use const whenever possible